### PR TITLE
build.py : Use `podman`

### PR DESCRIPTION
**Pull request by johnhaddon**
This is what we're using to build the container in CI, and is generally a bit nicer than Docker. One big benefit is that you don't need to run a daemon to use `podman`, particularly since the Docker daemon breaks the licensing for a certain piece of software that might be a little bit important at the moment.

### CODE REVIEW

File: build.py [hunk 1]
```diff
@@ -304,13 +299,13 @@ def releaseId() :
 # Here we're actually doing the build, this will run either locally or inside
 # the container bootstrapped above
 
-if os.path.exists( "/.dockerenv" ) and args.project == "gaffer" :
+if os.path.exists( "/run/.containerenv" ) and args.project == "gaffer" :
 
 	# Start an X server so we can generate screenshots when the
 	# documentation builds.
-	os.system( "Xvfb :99 -screen 0 1280x1024x24 &" )
```
Comment (murraystevenson) [inline build.py]: I'd suspect the addition of the `.0` to the display number here to be redundant as the screen number is already being specified by `-screen 0`. Did you find this necessary?

**Review by murraystevenson — APPROVED** [review]
Nice to continue the move away from Docker, and I like the switch to the more generic `container` in the argument names. One very minor question inline, but nothing that would hold this up from being merged.

File: build.py (hunk 1 above)
Comment (johnhaddon) [reply build.py]: You're completely right. I was floundering around trying to get the docs build to work, and matched this to the CI in the hope that was the problem. But it wasn't - it was the `.dockerenv` vs `.containerenv` difference that was the problem. And I was too lazy to revert the X changes and wait for another build to see if they were also necessary or superfluous. I've done that test now, and reverted the unnecessary changes in the squashed https://github.com/GafferHQ/build/pull/43/commits/e3a7cf03c5bf3785845d334cd00ef6a6463f7b0a. Thanks for pointing that out.

**Comment by murraystevenson** [comment]
Thanks for the changes, I've confirmed that I can build a package including docs on my system with the latest push. Merging.

**Pull request merged**